src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -463,85 +467,118 @@ private Type EmitIListCoercion<T>(AssociativeNode arg)
             {
-                mbs = mBase;
+                return mBase;
             }
-            else
+            
+            if (!CoreUtils.IsInternalMethod(methodName))
+            {
+                int cid = runtimeCore.ClassTable.GetClassId(className);
+                Validity.Assert(cid != DSASM.Constants.kInvalidIndex);
+            }
+
+            // TODO_MSIL: Figure out polymorfism when calling functions
+            // That should be done at runtime (when we know the exact runtime type of the caller type)
+            var fg = runtimeCore.GetFuncGroup(methodName, className);
+
+            // This checks if there is a static property like Point.X(arg) 
+            // and if so renames it to Point.get_X(arg) so that it can be 
+            // found as a static getter in the class declaration.
+            if (fg == null && args.Count == 1)
+            {
+                // Try to find a getter
+                methodName = ProtoCore.DSASM.Constants.kGetterPrefix + methodName;
+                fg = runtimeCore.GetFuncGroup(methodName, className);
+            }
+
+            Validity.Assert(fg != null, "Did not find a function group");
+
+            foreach (var funcEnd in fg.FunctionEndPoints)
             {
-                if (!CoreUtils.IsInternalMethod(methodName))
+                var procNode = funcEnd.procedureNode;
+                Validity.Assert(procNode != null, "Expected to have a valid procedureNode");
+
+                bool isStaticOrConstructorOrInternal = procNode.IsStatic || procNode.IsConstructor || CoreUtils.IsInternalMethod(methodName);
+
+                bool matchingArgCount = isStaticOrConstructorOrInternal ?
+                    procNode.ArgumentTypes.Count == args.Count :
+                    (procNode.ArgumentTypes.Count + 1) == args.Count;
+
+                if (!matchingArgCount) {
+                    continue;
+                }
+
+                FFIMemberInfo fFIMemberInfo = null;
+                System.Type declaringType = null;
+                ParameterInfo[] parameterInfos = null;
+
+                if (CoreUtils.IsInternalMethod(methodName))//equivalent to if (funcEnd is JILFunctionEndPoint)
                 {
-                    var modules = ProtoFFI.DLLFFIHandler.Modules.Values.OfType<ProtoFFI.CLRDLLModule>();
-                    var assemblies = modules.Select(m => m.Assembly ?? (m.Module?.Assembly)).Where(m => m != null);
-                    foreach (var asm in assemblies)
+                    var mInfo = BuiltIn.GetInternalMethod(methodName);
+                    parameterInfos = mInfo.GetParameters();
+                    declaringType = mInfo.DeclaringType;
+                    fFIMemberInfo = new FFIMethodInfo(mInfo);
+                }
+                else if (funcEnd is FFIFunctionEndPoint ffiFep)
+                {
+                    FFIHandler handler = FFIFunctionEndPoint.FFIHandlers[ffiFep.activation.ModuleType];
```

Comment:
is this ever anything but clr ffi?

Reply:
that is what I've seen during testing. 
Not sure what other edge cases there could be